src/NLog.StructuredLogging.Json/LoggerExtensions.cs

```diff
@@ -91,11 +93,27 @@ private static void TransferDataToLogEventProperties(LogEventInfo log, object lo
                 return;
             }
 
-            logProperties.GetType()
-                .GetProperties()
-                .Where(p => p.GetIndexParameters().Length == 0)
-                .ToList()
-                .ForEach(x => log.Properties.Add(x.Name, x.GetValue(logProperties)));
+            if (IsDictionary(logProperties))
+            {                
+                foreach (var logProperty in (IDictionary<object, object>) logProperties)
```

Comment:
So we turn the keys to string anyway here: https://github.com/justeat/NLog.StructuredLogging.Json/blob/master/src/NLog.StructuredLogging.Json/Helpers/Mapper.cs#L60

Using a key that isn't a string  is really not recommended, but it looks like this code treating key as `object` isn't introducing any issues.